benches/end2end.rs

```diff
@@ -0,0 +1,273 @@
+use blstrs::Scalar as Fr;
+use criterion::{black_box, criterion_group, criterion_main, Criterion, SamplingMode, BenchmarkId};
+
+use lurk::{
+    eval::{
+        empty_sym_env,
+        lang::{Coproc, Lang},
+        Evaluator,
+    },
+    field::LurkField,
+    proof::nova::NovaProver,
+    proof::Prover,
+    ptr::Ptr,
+    store::Store,
+};
+use std::time::Duration;
+
+const DEFAULT_REDUCTION_COUNT: usize = 10;
+fn go_base<F: LurkField>(store: &mut Store<F>, a: u64, b: u64) -> Ptr<F> {
+    let program = format!(
+        r#"
+(let ((foo (lambda (a b)
+              (letrec ((aux (lambda (i a x)
+                               (if (= i b)
+                                     x
+                                     (let ((x (+ x a))
+                                            (a (+ a (* b 2))))
+                                       (aux (+ i 1) a x))))))
+                       (let ((x (+ (* a b) 4)))
+                         (aux 0 a x))))))
+  (foo {a} {b}))
+"#
+    );
+
+    store.read(&program).unwrap()
+}
+
+fn end2end_benchmark(c: &mut Criterion) {
+    let mut group = c.benchmark_group("end2end_benchmark");
+    group.sampling_mode(SamplingMode::Flat)
+         .measurement_time(Duration::from_secs(120))
+         .sample_size(10);
+    
+    let limit = 1_000_000_000;
+    let lang_vesta = Lang::<pasta_curves::Fq, Coproc<pasta_curves::Fq>>::new();
+    let reduction_count = DEFAULT_REDUCTION_COUNT;
+
+    // setup
+    let mut store = Store::default();
+    let env = empty_sym_env(&store);
+    let prover = NovaProver::new(reduction_count, lang_vesta.clone());
+
+    // use cached public params
+    let pp = fcomm::public_params(reduction_count).unwrap();
+
+    let size = (10, 0);
+    let benchmark_id = BenchmarkId::new("end2end_go_base_nova", format!("_{}_{}", size.0, size.1));
+
+    group.bench_with_input(benchmark_id, &size, |b, &s| {
+        b.iter(|| {
+            let ptr = go_base::<pasta_curves::Fq>(&mut store, s.0, s.1);
+            let _result = prover
+                .evaluate_and_prove(&pp, ptr, env, &mut store, limit, &lang_vesta)
+                .unwrap();
+        })
+    });
+    
+    group.finish();
+}
+
+fn store_benchmark(c: &mut Criterion) {
```

Comment:
These `store`/`hydration` benches are really fast, only taking around 300 microseconds for each call. Should we try to stress test with bigger files? e.g. Lurk files with 1-10k LOC to see how it scales.